### DISCUSSION TITLE
How to translate the Smiles string 'CN=C(O)C' to ''CNC(=O)C' ?

### DISCUSSION THREAD

**Discussion by ZhitaoTian** [rdkit/rdkit#5333] [Q&A] [answered]
How recognize the conjugated double bonds, such as a peptide bond?
How to translate the Smiles string 'CN=C(O)C' to ''CNC(=O)C' ? 
The below-script seems not work
```python
a = Chem.MolFromSmiles("CCCCCCCC(=O)C(C)N=C(O)CCC(=O)O")
sub1=Chem.MolFromSmarts("N=C([OH1])C")
sub2=Chem.MolFromSmiles("NC(=[O])C")
Chem.ReplaceSubstructs(a,sub1,sub2,replaceAll=True)[0]
```
the result:
![image](https://user-images.githubusercontent.com/46099997/170627464-ea499154-048c-44a6-807f-97aaa34a10bb.png)

**Comment by wopozka** [comment, marked as answer]
You can use reaction for this purpose, eg:

```
mol = Chem.MolFromSmiles('CCCCCCCC(=O)C(C)N=C(O)CCC(=O)O')
reaction = AllChem.ReactionFromSmarts('[CX4,c:1][NX2:2]=[CX3:3]([OH:4])[CX4,c:5]>>[*:1][NH:2][C:3](=[O:4])[*:5]')
res = reaction.RunReactants([mol])
print(Chem.MolToSmiles(res[0][0]))
```
`CCCCCCCC(=O)C(C)NC(=O)CCC(=O)O`

**Reply by ZhitaoTian** [reply]
Thanks a lot, but it cannot deal with the molecule with multiple  peptide bonds.
```python
 mol = Chem.MolFromSmiles('CCCCN=C(O)CCCC(=O)C(C)N=C(O)CCC(=O)O')
```

**Reply by wopozka** [reply]
@ZhitaoTian 
```
mol = Chem.MolFromSmiles('CCCCN=C(O)CCCC(=O)C(C)N=C(O)CCC(=O)O')
reaction = AllChem.ReactionFromSmarts('([CX4,c:1][NX2:2]=[CX3:3]([OH:4])[CX4,c:5].[CX4,c:11][NX2:12]=[CX3:13]([OH:14])[CX4,c:15])>>([*:1][NH:2][C:3](=[O:4])[*:5].[*:11][NH:12][C:13](=[O:14])[*:15])')
res = reaction.RunReactants([mol])
print(Chem.MolToSmiles(res[0][0]))
```
`CCCCNC(=O)CCCC(=O)C(C)NC(=O)CCC(=O)O`
This is just example how you can do it, and up to you to extend the idea further. You can also do another thing, run the first reaction in loop until no reaction center is present.
